pulp_deb/app/tasks/synchronizing.py

```
@@ -263,6 +279,16 @@ async def run(self):
                     log.debug("Codename: {}".format(release_file.codename))
                     log.debug("Components: {}".format(release_file.components))
                     log.debug("Architectures: {}".format(release_file.architectures))
+
+                    if "Release" or "InRelease" in da_names:
```

This seems a little bit inconsistent. If we call for a release and don't find it, we ignore it, but if one of the releases we found lacks a component or architecture, we fail hard.

Also i see a real problem in that architectures get added and removed between Debian releases. And this change would prevent to specify a subset of architectures when synching multiple releases, and one of the arch's is missing from one of the releases.

---

So I guess these really are two separate points:

1. We should be consistent. If we are going to "validate" `components` and `architectures`, we should also validate `distributions` (releases) in the same way.
2. There is a conflict between preserving the valid (but perhaps not recommended) use case of wanting to recreate multiple upstream `distributions` (within a single pulp repository), while at the same time filtering/restricting to for example, a subset of the available architectures (if present) on the one hand, and wanting to catch bad user input on the other hand.

I do think that being maximally permissive with user input (and simply silently publishing a bunch of empty `distributions`, `compnents`, or `architectures`) whenever users make mistakes, opens the door to a lot of hard to diagnoze problems for most users to preserve a pretty rare specialist use case.

We could try to mitigate this with documentation (but most users wont read it).

I will change the status of this PR to draft until we have had a chance to discuss this further.

---

I guess it comes down to: Do we want it to be a valid use case to publish many distributions in a single pulp repository, while filtering the components and/or architectures, even if not all of the upstream distributions have all of the components/architectures we are filtering for.

---

And i think the answer is yes.
BTW: "<...> even if not all of the upstream distributions have all of the components/architectures we are filtering for." that is the part we have no control over.
Another, much more involved solution might be to explicitly specify the components to sync from per release requested.

If we want to keep the current, easy to grasp, configuration model, we could fail hard only on not found releases, to turn the inconsistency in the opposite direction. I guess that would catch most configuration errors, while preserving all the filtering features for arch/comp.